packages/expo-modules-core/ios/Tests/PersistentFileLogSpec.swift

```diff
@@ -0,0 +1,81 @@
+import ExpoModulesTestCore
+
+@testable import ExpoModulesCore
+
+class PersistentFileLogSpec: ExpoSpec {
+  let serialQueue = DispatchQueue(label: "dev.expo.modules.test.persistentlog")
+
+  let log = PersistentFileLog(category: "dev.expo.modules.test.persistentlog")
+
+  func clearEntriesSync() {
+    serialQueue.sync {
+      let sem = DispatchSemaphore(value: 0)
+      log.clearEntries { _ in
+        sem.signal()
+      }
+      sem.wait()
+    }
+  }
+
+  func filterEntriesSync(filter: @escaping PersistentFileLogFilter) {
+    serialQueue.sync {
+      let sem = DispatchSemaphore(value: 0)
+      log.filterEntries(filter: filter) { _ in
+        sem.signal()
+      }
+      sem.wait()
+    }
+  }
+
+  func appendEntrySync(entry: String) {
+    serialQueue.sync {
+      let sem = DispatchSemaphore(value: 0)
+      log.appendEntry(entry: entry) { _ in
+        sem.signal()
+      }
+      sem.wait()
+    }
+  }
```

Comment:
Let's move the helpers to be after the `spec` function. It's always better to see the main/public things on top and internal/private things at the end.